File: backend/app/models/matching.py
from pydantic import BaseModel


class MatchScoreBreakdown(BaseModel):
    """Core hybrid-score output — field names match the notebook's
    compute_hybrid_scores_batch() dict keys exactly (final_score,
    semantic_score, skills_score, experience_score, matched_skills,
    missing_skills), so anyone comparing this API's output against the
    notebook's own evaluate_resume_pretty() report can do so directly."""

    final_score: float
    semantic_score: float
    skills_score: float
    experience_score: float
    matched_skills: list[str]
    missing_skills: list[str]


class DriveMatchScoreResponse(MatchScoreBreakdown):
    """Phase 7: single resume x single drive."""

    drive_id: str


class RecommendedDriveResponse(MatchScoreBreakdown):
    """Phase 9: one row of the student's ranked drive recommendations."""

    drive_id: str
    job_title: str
    company_name: str
    location: str | None
    package: str | None


class RankedApplicantResponse(MatchScoreBreakdown):
    """Phase 9: one row of a TPO's ranked applicant list for a drive."""

    application_id: str
    student_id: str
    student_name: str
    student_department: str | None = None
    student_cgpa: float | None = None
    resume_id: str
    resume_filename: str | None = None
    status: str = "applied"
    # Eligibility
    eligibility_passed: bool | None = None
    eligibility_reasons: list[str] = []
    # Assessment
    assessment_status: str = "not_required"
    assessment_score_pct: float | None = None
    # Decision
    rejection_reasons: list[str] = []
    rejection_note: str | None = None
    decision_at: str | None = None
    applied_at: str | None = None

File: frontend/types/matching.ts
export interface MatchScoreBreakdown {
  final_score: number;
  semantic_score: number;
  skills_score: number;
  experience_score: number;
  matched_skills: string[];
  missing_skills: string[];
}

export interface DriveMatchScore extends MatchScoreBreakdown {
  drive_id: string;
}

export interface RecommendedDrive extends MatchScoreBreakdown {
  drive_id: string;
  job_title: string;
  company_name: string;
  location: string | null;
  package: string | null;
}

export interface RankedApplicant extends MatchScoreBreakdown {
  application_id: string;
  student_id: string;
  student_name: string;
  student_department?: string | null;
  student_cgpa?: number | null;
  resume_id: string;
  resume_filename?: string | null;
  status: string;
  eligibility_passed?: boolean | null;
  eligibility_reasons?: string[];
  assessment_status?: string;
  assessment_score_pct?: number | null;
  rejection_reasons?: string[];
  rejection_note?: string | null;
  decision_at?: string | null;
  applied_at?: string | null;
}
